core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/autotelemetry21/custom_telemetry.go

```diff
@@ -121,6 +125,9 @@ func (e *AutomationCustomTelemetryService) Close() error {
 }
 
 func (e *AutomationCustomTelemetryService) sendNodeVersionMsg() {
+	e.mu.RLock()
+	defer e.mu.RUnlock()
```

Comment:
Does the lock need to be held while doing the blocking send? Or only to copy `e.configDigest`?

Reply:
Oh I thought we'd need the read lock for both send functions given that they're each called from different go routines; are you suggesting we could instead use a `sync.Mutex` and lock only the write operation?

Reply:
I mean can we reduce the scope of the lock to just reading the `configDigest` value:
```suggestion
	e.mu.RLock()
	configDigest := e.configDigest
	e.mu.RUnlock()
```

Reply:
Oh sorry I totally wasn't following; yes that makes sense!